src/os/structs/structs.js

```diff
@@ -0,0 +1,21 @@
+goog.provide('os.structs');
+
+
+/**
+ * Flattens a tree into a specified array - optionally filtering.
+ * @param {os.structs.ITreeNode} node
+ * @param {Array<os.structs.ITreeNode>} results
+ * @param {function(os.structs.ITreeNode): boolean=} opt_filter
+ */
+os.structs.flattenTree = function(node, results, opt_filter) {
+  opt_filter = opt_filter || goog.functions.TRUE;
+  if (opt_filter(node)) {
+    results.push(node);
+    var children = node.getChildren();
```

Comment:
This only recurses if the result of the filter is `true`. Was that your intention? I don't think that is correct.

Reply:
Thanks! Fixed that